USING_DOCKER.md

```diff
@@ -0,0 +1,137 @@
+before each new commit to git, install the `pre-commit` program
+on the machine that hosts your Docker container.
+Then (one time) run `pre-commit install-hooks` to set up
+your `git` client to use the `.pre-commit-config.yaml` config file.
+These tests will run each time you commit so you can avoid silly mistakes.
+(NB: `pre-commit` must be done on the host machine
+since the Docker container only reads the repository.)
+
+Finally, start up the Docker container with:
+
+```bash
+cd <top-level-PRQL-directory>
+docker run -it -v $(pwd)/:/app -p 3000:3000 prql
+```
+
+- You'll see a `root@xxxxxxxxx:/app/#` prompt
+- Enter the commands below for the component you're working on
+- Ctl-C to exit that component
+- Enter 'exit' to close down the Docker machine
+
+## Running components under Docker
+
+The following commands come from the `README.md`
+file in each of the component directories.
+Use the `docker run...` command above, then enter
+the command(s) below.
+
+_(The first time you run a component, it may take some
+time to install additional files.
+Once they're built, start up is quick.)_
+
+**Playground:** Use the command above, then enter:
+
+```bash
+cd playground
+npm install # first time only
+npm start
+```
+
+**Language Book:** Use the command above, then enter:
+
+```bash
+cd book
+mdbook serve -n 0.0.0.0 -p 3000
+```
+
+**Website:** Use the command above, then enter:
+
+```bash
+cd website
+hugo server --bind 0.0.0.0 -p 3000
+```
+
+**prql-compiler:** Use the command above,
+`cd prql-compiler` then read the **Usage** section of the
+[README.md](./prql-compiler/README.md)
+
+**prql-java:** Use the command above,
+`cd prql-java` then read the **Usage** section of the [README.md](./prql-java/README.md)
+
+**prql-js:** Use the command above,
+`cd prql-js` then read the **Usage** section of the [README.md](./prql-js/README.md)
+
+**prql-lib:** Use the command above,
+`cd prql-lib` then read the **Usage** section of the [README.md](./prql-lib/README.md)
+
+**prql-macros:** Use the command above,
+`cd prql-macros` then read the **Usage** section of the [README.md](./prql-macros/README.md)
+
+**prql-python:** Use the command above,
+`cd prql-python` then read the **Usage** section of the [README.md](./prql-python/README.md)
+
+## Manual testing for Dockerfile
+
+While the Dockerfile is under development, use these minimal tests
+before committing new code.
+
+1. **Check the Taskfile.yml** Run these commands to ensure that the
+   `Taskfile.yml` still builds the "normal" environment:
+
+   ```bash
+   cd <directory-with-prql>
+   cargo test
+   task setup-dev
+   ```
+
+1. **Build the Docker container** as described above.
+
+1. **Quick tests for the Docker container**
+   Start the container (as described above),
+   then check the various components (also, as described above)
+
+1. **Examine the Github actions/workflows** for errors in
+   your own repo before pushing to the main `prql/prql` repo.
```

Comment:
(No great stress, though we generally encourage early PRs, no need to look at tests on a fork before making one. I'm also not sure this bullet is docker-specific. But it's fine!)

Reply:
OK. A difference in philosophy. I generally figure that I don't know what I'm doing (especially on a new project), and treat warnings as a cluebat.

And once the warnings are all suppressed/eliminated, I treat a new one as a bigger hint...

But once we get a few miles on this Docker thingie, then I'll feel better that it actually works properly, and we can remove [ much | most ] of this "manual test" section.

Reply:
Yes great, let's see how it goes and we'll iterate on these